src/dbcp/transform/ballot_ready.py

```diff
@@ -57,19 +61,44 @@ def _normalize_entities(ballot_ready: pd.DataFrame) -> dict[str, pd.DataFrame]:
         "frequency",
         "partisan_type",
     ]
-    # position_id == 156594 is the only position with two frequencies and reference_years
-    # Create a new index for it
-    new_index = ballot_ready.position_id.max() + 1
-    assert new_index not in ballot_ready.position_id
-    ballot_ready.loc[ballot_ready.race_id == 2020783, "position_id"] = new_index
-    assert (
-        (ballot_ready.groupby(position_pk_fields)[position_fields].nunique() <= 1)
-        .all()
-        .all()
-    ), "There is duplicate entity information in the positions dataframe."
+    # A small number of positions have multiple records where frequencies and referece_year
+    # differ. This is because of edge cases where a position has changed its election year
+    # or frequency due to redistricting or an election law. The old position_election_frequency
+    # and the future/current one are reflected in the data.
+
+    # check if position_id is unique
+    deduplicated_positions = ballot_ready[position_fields].drop_duplicates()
+    is_duplciate_position = deduplicated_positions.position_id.duplicated(keep=False)
+
+    # If position_id is not unique, generate new position ids for the duplicates
+    if is_duplciate_position.any():
+        n_duplicates = is_duplciate_position.sum()
+        logger.info(
+            f"{n_duplicates / len(deduplicated_positions) * 100} % of positions have duplicates. Generating new position_ids for duplicates."
+        )
+
+        # Create a mapping of the old position_ids to the new position_ids
+        new_index = ballot_ready.position_id.max() + 1
+        duplicate_position_ids = deduplicated_positions[
+            is_duplciate_position
+        ].position_id.to_frame()
+        duplicate_position_ids["new_position_id"] = range(
+            new_index, new_index + n_duplicates
+        )
+        new_position_id_mapping = duplicate_position_ids.set_index(
+            "position_id"
+        ).to_dict()["new_position_id"]
+
+        ballot_ready["position_id"] = ballot_ready["position_id"].replace(
+            new_position_id_mapping
+        )
+
```

Comment:
This doesn't feel great to me. I think a better option might be to break out the one to many relationship between `position_id` and `frequency` and `reference_year` into another table. What do you think @TrentonBush